pages/[username].tsx

```diff
@@ -0,0 +1,198 @@
+                    {data?.allow_anon_dm ? (
+                        <>
+                            <div className="flex flex-grow items-ce bg-gray05 rounded-xl py-1 px-2 items-center">
+                                <textarea ref={textAreaRef} rows={1} onChange={(e) => {
+                                    setMessage(e.target.value)
+                                }} className="bg-transparent min-h-[24px] w-full"
+                                    style={{ resize: 'none' }}
+                                    placeholder="Send me a message 😀" />
+                                <Toggle
+                                    checked={true}
+                                    className="bg-white"
+                                    defaultChecked={true}
+                                    onClick={() => showDownloadToast()}
+                                    icons={{
+                                        checked: <text className="text-xs font-normal text-cyan leading-[18px]">On</text>,
+                                        unchecked: null,
+                                    }}
+                                />
+                            </div>
+                            <button className="rounded-full flex-shrink-0 bg-cyan h-8 w-8 flex items-center justify-center" onClick={() => {
+                                localStorage.setItem(MessageEnum.tempMessage, message);
+                                router.push('/verification')
+                            }}>
+                                <Image src='/image/sendMessageicon.svg' alt="icon send" width={32} height={32} />
+                            </button>
+                        </>
+                    ) : (
+                        <>
+                            <div className="flex flex-grow bg-gray05 rounded-xl py-1 px-2">
+                                <div className="min-h-[24px] w-[230px] text-sm">
+                                This user does not want to receive incognito messages.
+                                </div>
+                            </div>
+                        </>
+                    )}
+                </div>}
+            </LayoutContainer>
+        </BaseContainer>
+    )
+}
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+    const { username } = context.query; // Retrieve the URL parameter from context.query
+    let userAgent = parser(context?.req?.headers['user-agent'])
+
+    let isDynamicLink = username?.indexOf('+') > -1
+    let originalUsername = !isDynamicLink ? username : username?.slice(0, username?.length - 2)
+
+    if (UserAgentUtils.isMobile(userAgent)) {        
+        let redirect = await RedirectUtils.redirectUsernameForMobileDevice(userAgent, username)
+        if (redirect && !isDynamicLink) return redirect
+    }
+
+    return {
+        props: {
+            username: originalUsername,
+            isDynamicLink
+        }
+    }
+
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getServerSideProps` function, there's no error handling for the case where `RedirectUtils.redirectUsernameForMobileDevice` fails. It would be good to add a try-catch block around this call to handle any potential errors.
```diff
    if (UserAgentUtils.isMobile(userAgent)) {        
        let redirect = await RedirectUtils.redirectUsernameForMobileDevice(userAgent, username)
        if (redirect && !isDynamicLink) return redirect
    }
+    if (UserAgentUtils.isMobile(userAgent)) {
+        try {
+            let redirect = await RedirectUtils.redirectUsernameForMobileDevice(userAgent, username)
+            if (redirect && !isDynamicLink) return redirect
+        } catch (error) {
+            console.error('Failed to redirect:', error);
+        }
+    }
```

Lastly, it's a good practice to add a newline at the end of the file to avoid any potential issues with version control systems or other tools that expect or require it.
```diff
};
\ No newline at end of file
+};
+
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->